### PR TITLE
Updating caseflow service class

### PR DESCRIPTION
**Pull request by raymond-hughes**
This change refactors CaseflowService class to remove some coupled code allowing a more general usecase

### CODE REVIEW

File: app/services/external_api/caseflow_service.rb [hunk 1]
```diff
@@ -4,67 +4,57 @@
 class ExternalApi::CaseflowService
   include LoggerMixin
   # Base endpoint for the Caseflow API events.
-  BASE_ENDPOINT = "api/events/v1/"
+  BASE_ENDPOINT = "api/events/v1".freeze
```
Comment (gcd253) [inline app/services/external_api/caseflow_service.rb:7]: The linter identifies this freeze as an issue, with the following message:
`Do not freeze immutable objects, as freezing them has no effect.`

**Review by dcoleman21 — DISMISSED** [review]
The code looks a lot more dynamic

File: spec/services/external_api/caseflow_service_spec.rb [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
 describe ExternalApi::CaseflowService do
```
Comment (gcd253) [inline spec/services/external_api/caseflow_service_spec.rb:3]: Is there any explicit testing for the methods `establish_person_updated_records_from_event!` and `establish_person_updated_event_error!`? I don't think I see any, but would suggest including some

Comment (gcd253) [reply spec/services/external_api/caseflow_service_spec.rb:3]: Resolved with latest code change

**Review by dcoleman21 — APPROVED** [review]
Nice job de-coupling some of this.

File: app/models/virtual/transformers/person_updated.rb [hunk 3]
```diff
@@ -14,7 +14,6 @@ class Transformers::PersonUpdated
     "first_name" => [String, NilClass],
     "middle_name" => [String, NilClass],
     "last_name" => [String, NilClass],
-    "email_address" => [String, NilClass],
```
Comment (dcoleman21) [inline app/models/virtual/transformers/person_updated.rb:17]: Did we remove email address because it is not necessary for a person objec?

**Review by dcoleman21 — COMMENTED** [review]
See my comment about the email

**Review by gcd253 — APPROVED** [review]
Looking good, excellent work!

**Review by dcoleman21 — APPROVED** [review]

**Pull request merged**